dev/release/verify-release-candidate.sh

```diff
@@ -0,0 +1,132 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+case $# in
+  2) VERSION="$1"
+     RC_NUMBER="$2"
+     ;;
+  *) echo "Usage: $0 X.Y.Z RC_NUMBER"
+     exit 1
+     ;;
+esac
+
+set -e
+set -x
+set -o pipefail
+
+SOURCE_DIR="$(cd "$(dirname "${BASH_SOURCE[0]:-$0}")" && pwd)"
+COMET_DIR="$(dirname $(dirname ${SOURCE_DIR}))"
+COMET_DIST_URL='https://dist.apache.org/repos/dist/dev/datafusion'
+
+download_dist_file() {
+  curl \
+    --silent \
+    --show-error \
+    --fail \
+    --location \
+    --remote-name $COMET_DIST_URL/$1
+}
+
+download_rc_file() {
+  download_dist_file apache-datafusion-comet-${VERSION}-rc${RC_NUMBER}/$1
+}
+
+import_gpg_keys() {
+  download_dist_file KEYS
+  gpg --import KEYS
+}
+
+if type shasum >/dev/null 2>&1; then
+  sha256_verify="shasum -a 256 -c"
+  sha512_verify="shasum -a 512 -c"
+else
+  sha256_verify="sha256sum -c"
+  sha512_verify="sha512sum -c"
+fi
+
+fetch_archive() {
+  local dist_name=$1
+  download_rc_file ${dist_name}.tar.gz
+  download_rc_file ${dist_name}.tar.gz.asc
+  download_rc_file ${dist_name}.tar.gz.sha256
+  download_rc_file ${dist_name}.tar.gz.sha512
+  verify_dir_artifact_signatures
+}
+
+verify_dir_artifact_signatures() {
+  # verify the signature and the checksums of each artifact
+  find . -name '*.asc' | while read sigfile; do
+    artifact=${sigfile/.asc/}
+    gpg --verify $sigfile $artifact || exit 1
+
+    # go into the directory because the checksum files contain only the
+    # basename of the artifact
+    pushd $(dirname $artifact)
+    base_artifact=$(basename $artifact)
+    ${sha256_verify} $base_artifact.sha256 || exit 1
+    ${sha512_verify} $base_artifact.sha512 || exit 1
+    popd
+  done
+}
+
+setup_tempdir() {
+  cleanup() {
+    if [ "${TEST_SUCCESS}" = "yes" ]; then
+      rm -fr "${COMET_TMPDIR}"
+    else
+      echo "Failed to verify release candidate. See ${COMET_TMPDIR} for details."
+    fi
+  }
+
+  if [ -z "${COMET_TMPDIR}" ]; then
+    # clean up automatically if COMET_TMPDIR is not defined
+    COMET_TMPDIR=$(mktemp -d -t "$1.XXXXX")
+    trap cleanup EXIT
+  else
+    # don't clean up automatically
+    mkdir -p "${COMET_TMPDIR}"
+  fi
+}
+
+test_source_distribution() {
+  set -e
+  pushd core
+    RUSTFLAGS="-Ctarget-cpu=native" cargo build --release
+  popd
+  ./mvnw install -Prelease -DskipTests -P"spark-3.4" -Dmaven.gitcommitid.skip=true
```

Comment:
Spark-3.4 is hardcoded now. What about making it a parameter.

Also does it make sense to use one of our make commands

Reply:
We don't currently have a make command that specifies `-Dmaven.gitcommitid.skip=true` although one is being added in https://github.com/apache/datafusion-comet/pull/512

Reply:
Right, one hack could be something like `make release PROFILES="-Dmaven.gitcommitid.skip=true"`

Reply:
Should we run tests as well? All comet tests pass for me when I run them though they do take a while to run.

Reply:
> Right, one hack could be something like `make release PROFILES="-Dmaven.gitcommitid.skip=true"`

That _would_ be a hack. I would leave it as it is. I assume that we will update/deprecate some of this once we move to publishing binary releases.

Reply:
> Should we run tests as well? All comet tests pass for me when I run them though they do take a while to run.

I think that the tests take too long. We create the release candidate tarball from a tag in GitHub where we have already run all of the tests, so running them all again is potentially redundant.

Perhaps there is some subset of tests that we could run? 🤔

Reply:
> Spark-3.4 is hardcoded now. What about making it a parameter.

I added a comment explaining that we are testing the latest supported Spark version.

I think we can make more improvements in the future but I would like to get the basic scripts in place first so that we have the ability to make a release.